library/src/ui/index.js

```diff
@@ -1 +1,72 @@
-window.CoNDeT.ui = {}
\ No newline at end of file
+window.CoNDeT.ui = {
+    BaseComponent: {
+        state: {
+
+        },
+        common: {
+
+        },
+        init: function(common, props) {
+            this.ref = this.onInit();
+        },
+        update: function(common, props) {
+            children.update();
+            this.onUpdate();
+            this.ref.onUpdate();
+        },
+        destroy: function(common, props) {
+            children.destroy();
+            this.onDestroy();
+            this.ref.onDestroy();
+        },
+        getPosition: function() {
+            return {
+                x: this.domReference.offsetLeft,
+                y: this.domReference.offsetTop
+            };
+        },
+        getDimensions: function() {
+            return {
+                width: this.domReference.clientWidth,
+                height: this.domReference.clientHeight
+            }
+        },
+        containsPoint: function(x, y) {
+            return (x >= this.domReference.offsetLeft && 
+                x <= this.domReference.offsetLeft + this.domReference.clientWidth &&
+                y >= this.domReference.offsetTop &&
+                y <= this.domReference.offsetTop + this.domReference.clientHeight) ? true : false;
+        },
```

Comment:
And again: `domReference` -> `ref`  
Dodatkowo używanie ternary tutaj wydaje się mało czytelne. W JS operatory logicznie nie zwracają boolean (no może poza `!`) więc lepiej jawnie podawać typy. Ale tutaj masz same porównania które już zwracają `boolean`. Więc możne zwrócić efekt operatorów logiczny bez dodawania `? true : false`  
Co do poprawności algorytmu, to ufam, że działa 😜